runtime/jvmti/jvmtiThread.c

```diff
@@ -665,28 +663,50 @@ jvmtiGetOwnedMonitorInfo(jvmtiEnv* env,
 		ENSURE_NON_NULL(owned_monitors_ptr);
 
 		rc = getVMThread(currentThread, thread, &targetThread, TRUE, TRUE);
-		if (rc == JVMTI_ERROR_NONE) {
-			jobject * locks;
+		if (JVMTI_ERROR_NONE == rc) {
+			jobject *locks = NULL;
 			jint count = 0;
+			J9VMThread *threadToWalk = targetThread;
 
-			vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
+#if JAVA_SPEC_VERSION >= 19
+			J9VMThread stackThread = {0};
+			J9VMEntryLocalStorage els = {0};
+			j9object_t threadObject = (NULL == thread) ? currentThread->threadObject : J9_JNI_UNWRAP_REFERENCE(thread);
+			J9VMContinuation *continuation = getJ9VMContinuationToWalk(currentThread, targetThread, threadObject);
+			if (NULL != continuation) {
+				vm->internalVMFunctions->copyFieldsFromContinuation(currentThread, &stackThread, &els, continuation);
+				threadToWalk = &stackThread;
+			}
+#endif /* JAVA_SPEC_VERSION >= 19 */
 
-			count = walkLocalMonitorRefs(currentThread, NULL, targetThread, UDATA_MAX);
+#if JAVA_SPEC_VERSION >= 19
+            if (NULL != targetThread)
+#endif /* JAVA_SPEC_VERSION >= 19 */
+			{
+				vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
+			}
```

Comment:
Looks from the code we are still walking an unmounted virtual thread ? just not calling `haltThreadForInspection`
This seems unnecessary as unmounted virtual thread should not have any owned monitors otherwise it should not be allowed to unmount due to pinning